### ISSUE TITLE
rigl_constant_fan.py -> _get_grow_mask improvements

### ISSUE THREAD

**Issue by mklasby** [calgaryml/condensed-sparsity#8]
> I'm not convinced iterating through all indices of mask is the best way of doing this - that's fine for now, correctness before speed, but maybe put a TODO and bug for this to investigate it once we are sure this works well.

I think we want to investigate using topk,, torch.where and boolean matrix operations to create this this. Would have to collect topk for every row, however even if we have to loop over rows this is going to be better than looping over indices though. Probably have missed something, but pseudo-code of what I'm thinking:

unused_saliency = score_grow*~drop_mask
for row in num_rows:
    least_salient = torch.topk(unused_saliency[row], k=fan_in, sorted=True)[-1]
    grow_mask = score_grow >= least_salient
Use torch.topk on unused saliency rows to figure out the smallest saliency that is eligible to 'grow', where k is essentially fan_in for every row (neuron)
use torch.where(saliency >= smallest saliency) to select mask of weights to grow in a row
